### PR TITLE
DCOS-8058: Add client side container validation

### PR DESCRIPTION
**Pull request by orlandohohmeier**
---
ℹ️  You will see this validation error only on the second click or after switching the tabs. This is know issue which will hopefully be fixed with https://github.com/mesosphere/reactjs-components/pull/303.

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @pierlo-upitup

File: src/js/schemas/service-schema/ContainerSettings.js [hunk 1]
```diff
@@ -24,6 +25,21 @@ const ContainerSettings = {
...
+          externalValidator: function ({containerSettings}, definition) {
+            const {image} = containerSettings;
+
+            if (image == null ||
+              ContainerValidatorUtil.isValidDockerImage(image)) {
+              return true;
+            }
+
+            definition.showError = 'Container Image, must not contain ' +
+              'whitespace, and should not contain any other characters ' +
```
Comment (aldipower) [inline src/js/schemas/service-schema/ContainerSettings.js]: Not sure if the commas behind "Container Image" and "whitespace" and "underscores" are placed correctly.

**Comment by aldipower** [comment]
LGTM!

File: src/js/schemas/service-schema/ContainerSettings.js (hunk 1 above)
Comment (orlandohohmeier) [inline src/js/schemas/service-schema/ContainerSettings.js]: That happens, if you copy paste stuff... 🍝 😇  I'll remove them.

**Comment by orlandohohmeier** [comment]
**mesosphere-ci** retest this please

File: src/js/utils/ContainerValidatorUtil.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const ContainerValidatorUtil = {
+  isValidDockerImage(dockerImage) {
+    if (typeof dockerImage !== 'string' || dockerImage === '') {
+      return false;
+    }
+
+    // The pattern is based on Dockers `validContainerName` regexp:
+    // https://github.com/docker/docker/blob/f63cdf0260/runtime.go#L33
+    return /^[a-zA-Z0-9_\-/:.]+$/.test(dockerImage);
```
Comment (rcorral) [inline src/js/utils/ContainerValidatorUtil.js]: The pattern isn't the same. Does it matter? ¯\\_(ツ)_/¯

**Comment by rcorral** [comment]
Looks pretty good aside from my one comment. Feel free to merge if you feel good about the regex.

File: src/js/utils/ContainerValidatorUtil.js (hunk 2 above)
Comment (orlandohohmeier) [inline src/js/utils/ContainerValidatorUtil.js]: It's based (!) on the `validContainerName` pattern but also allows `/`,`:`, and `.` which are used in the other parts (user/org and tag) of a valid docker image specification.

**Pull request merged**